### ISSUE TITLE
Feature request: Support mixed color/B&W DHGR mode

### ISSUE THREAD

**Issue by bhtooefr** [rowokii/A2C_TangyPack#1]
Video7 and Le Chat Mauve sold adapters for the //c that provided an RGB output, and added a new video mode, a so-called "mix mode", using the unused high bit of every video byte to switch between color and monochrome.

As far as I'm aware, no modern video adapters for the //c support this mode.

Documentation from adapter manufacturers:
Video7 Color Enhancer IIc (for CGA-style monitors): https://mirrors.apple2.org.za/Apple%20II%20Documentation%20Project/Peripherals/Video/Video-7%20Color%20Enhancer%20IIc/Manuals/Video-7%20Color%20Enhancer%20IIc%20-%20Manual.pdf
Video7 RGB Interface //c (for analog RGB monitors): https://mirrors.apple2.org.za/ftp.apple.asimov.net/documentation/hardware/video/Video7%20for%20Apple%20IIc%20Manual.pdf

A couple of //e interfaces' documentation:
Video7 64K RGB Interface //e (may have some programming info for the Video7 implementation, but also describes some modes not available on the //c - this adapter was also rebranded by Apple as the official RGB card for the //e in the US): https://mirrors.apple2.org.za/ftp.apple.asimov.net/documentation/hardware/video/Video-7%20RGB-SL7.pdf
Le Chat Mauve Feline (//e card that is compatible with the //c adapter): https://ftp.apple.asimov.net/documentation/hardware/video/Le%20chat%20mauve%20Feline%20manuel%20OCR.pdf

An example of mix mode on a NTSC //c, using a Le Chat Mauve IIc RVB Peritel adapter (intended for 50 Hz //cs, but worked perfectly on my 60 Hz), with a generic SCART to HDMI adapter, into a HDMI to VGA adapter to go into a 21" CRT:
![IMG_20210724_1201584](https://github.com/user-attachments/assets/fb8db5d5-d8d7-45ef-ba47-77370c8d86ca)

I'm unsure how these adapters actually detected selecting mix mode, or how they detected the high bit on the //c video port, but it's clear that they're doing it _somehow_.

**Comment by rowokii** [comment]
Do you have any software that generates the mixed mode image?  Maybe I can poke at the video port and see what's coming out when it's using this mode.

**Comment by bhtooefr** [comment]
http://mirrors.apple2.org.za/Apple%20II%20Documentation%20Project/Interface%20Cards/Apple%20IIe/Apple%20IIe%20Extended%2080%20Column%20RGB%20Card/Disk%20Images/ has some demo disks for the Video7 //e card.

From that card's manual:

![image](https://github.com/user-attachments/assets/0381d0ad-bd11-42a2-8369-739d6d42b99d)
![image](https://github.com/user-attachments/assets/932c3cea-1227-42bc-ab62-dbcefab08cfe)

http://boutillon.free.fr/Underground/Anim_Et_Graph/Extasie_Chat_Mauve/Dsp_Img/Disks/Dsp_Img.dsk.gz is targeted towards the Le Chat Mauve Feline and IIc adapter.

I'd also look at https://github.com/AppleWin/AppleWin/issues/764 as it discusses some differences between the implementations.

**Comment by rowokii** [comment]
OK, so my code tries to imitate the artifact coloring effect of the NTSC monitor/TV vs as far as I can tell, the RGB interface does a simple straight up map color to bit patterns.  So, in theory, it should be possible to map signal to match the RGB output, but in practice it might be a pain to try to figure out the actually mapping.   I'm going to have to back burner this for a bit while I work on couple of higher priority feature sets I have in mind, but I definitely want to explore this some more.   

My device displays pretty much what I see in Applewin when set to color monitor/TV mode. 

![PXL_20240728_042112442](https://github.com/user-attachments/assets/57276a66-76cf-4502-ac51-b7e8fde4d332)

What's strange is on the DHR.COLORS code, my output looks completely borked.

My Apple IIC 
![PXL_20240728_042257802](https://github.com/user-attachments/assets/efe37c50-3bba-4481-bd0e-0886e6964ff8)

AppleWin display of same demo
![applewin_cap1](https://github.com/user-attachments/assets/89d5b2ce-419f-460f-8928-97d373487f65)

If you look at the  Purple for example, my computer looks like it's completely missing the "P", "r", and "l".   ????

**Comment by bhtooefr** [comment]
That looks to me like the mode switching ended up in hires, not DHR?

In any case, I *think* an acceptable behavior would be to, essentially, disable color burst/NTSC color emulation on a per byte/7 pixel basis instead of a per frame basis? Probably not exactly accurate, but close enough. (You'll also run into, Le Chat Mauve colors are not quite the same as NTSC colors, and Video-7 colors range from "pretty close" to "literally mapping Apple II colors to CGA RGBI colors". I think ignoring *those* details would honestly be fine.)

Also, implementing the BW560 mode that disables color emulation per-frame automatically would be nice, too - some stuff like MouseDesk actually uses that if it's available. Might be easier to implement CO140 and BW560 first, as it's effectively just switching functionality that already exists on your adapter from the computer side rather than the buttons on it, then implement mix mode later.

**Comment by rowokii** [comment]
So, it looks like the DHR demo is not compatible with IIc.  It works fine in a IIe, but every IIc I have tried(rom 255, 0, 4 and IIc+), the initial drawing is fine, but it flicks at the end I get the distorted image.   

Looking at the signal available at the IIc video expansion port, I see a signal called "VIDD7 - From 74LS374. Video Latch.  Causes half-dot shift high"  This sure sounds like the 7th bit you are referring too.   Sounds a lot like the name of the interface Video-7 too! 

In any case, the bad news is that I will not be able to support this mode on the current hardware if I need to look at VIDD7 signal. 
1 - TEXT, 4 - SEGB, 6 - LDPS, 9 - PRAS,  & 14 - VIDD7 are signals that are not connected to the FPGA.  I would have to add another level shift buffer chip.

**Comment by rowokii** [comment]
Right.  Tangy Pack actually doesn't know what video mode the Apple IIc is in.  It just takes the pixel data and the color reference signal to map the artifact color and looks at the GR signal to see if it should suppress the color for the text screen (as well as in monochrome modes, of course).  So, for normal video modes, I don't need to look at VIDD7.  I think to get the mono in DHR mode, I will have to observe TEXT and GR signal (TEXT is inverse of GR signal, except when in DHR mode) and once I know we are in DHR, look at VIDD7 to change to mono.   So, I think it should be possible to support this, but once again, not with current hardware.  I should try to prototype this once have some time...
